enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -88,10 +88,14 @@ func (l *FileCacheLoader) currentSnapshotVersion(ctx context.Context, key *fcpb.
 	rn := digest.NewResourceName(versionKey, key.InstanceName, rspb.CacheType_AC, repb.DigestFunction_BLAKE3)
 	acResult, err := cachetools.GetActionResult(ctx, l.env.GetActionCacheClient(), rn)
 	if status.IsNotFoundError(err) {
-		// If the snapshot version has never been updated, version metadata
-		// won't exist in the cache. It's valid for the version ID to be empty
-		// in this case.
-		return "", nil
+		// Version metadata might not exist in the cache if:
+		// * The snapshot version has never been set (Ex. if you've never invalidated
+		//   a snapshot with this key)
+		// * The version metadata has expired from the cache
+		// In the latter case, we want to be careful to not fallback to an older,
+		// invalid snapshot. So here we generate a new version ID to guarantee
+		// we start from a clean snapshot.
+		return l.env.GetSnapshotService().InvalidateSnapshot(ctx, key)
```

Comment:
InvalidateSnapshot saves the version metadata in the cache. We want to make sure it's set, or every VM will generate a new version ID.

Ex. Let's say the version metadata expires from the cache.
T0: A VM starts up. There's no version metadata in the cache, so it generates a new version. It completes and saves a snapshot (saving a snapshot does not update the version metadata)
T1: Another VM starts up.  If we hadn't saved the version metadata at T0, this will also generate a new version